### DISCUSSION TITLE
Yarn SDKs not updated despite package.json picked up

### DISCUSSION THREAD

**Discussion by jpogorzelski** [renovatebot/renovate#16286] [Suggest an Idea]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

When using Yarn [SDKs with PnP installs](https://yarnpkg.com/getting-started/editor-sdks), renovatebot won't update the version in respective package.json

Example:
- Declare devDependency: `"prettier": "2.6.2"`
- run `yarn dlx @yarnpkg/sdks vscode`
- it generates the files in `<root>/.yarn/sdks/prettier` directory based on the version in `package.json`.  `prettier` sdk, there are two files created: `index.js` (JS module) and `package.json` with following content:
```
{
  "name": "prettier",
  "version": "2.6.2-sdk",
  "main": "./index.js",
  "type": "commonjs"
}
```

Renovate bot correctly picks up the new package.json file, but it doesn't find any dependencies in it. I suspect it is due to the fact that `name` and `version` are declared as root properties and not properties of "dependencies" object.

Example repo: https://github.com/jpogorzelski/renovate-yarn-with-sdks
Example PR: https://github.com/jpogorzelski/renovate-yarn-with-sdks/pull/3

As can be seen, the dependency in root `package.json` is picked up correctly and created PR for update to v2.7.1, the other one (SDK) remains unchanged.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Matched 4 file(s) for manager npm: .yarn/sdks/eslint/package.json, .yarn/sdks/prettier/package.json, .yarn/sdks/typescript/package.json, package.json
DEBUG: npm file .yarn/sdks/eslint/package.json has name "eslint"
DEBUG: Package file has no deps
DEBUG: npm file .yarn/sdks/prettier/package.json has name "prettier"
DEBUG: Package file has no deps

...
"npm": [
...
{
  "packageFile": ".yarn/sdks/prettier/package.json",
  "deps": [],
  "packageJsonName": "prettier",
  "packageFileVersion": "2.6.2-sdk",
  "packageJsonType": "library",
  "managerData": {
    "yarnZeroInstall": false,
    "hasPackageManager": false
  },
  "skipInstalls": true,
  "constraints": {
  }
}
   ...

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by viceice** [comment]
this is not a bug. renovate simply doesn't support yarn SDKs.

which command need to be run to update the yarn SDK?

**Reply by jpogorzelski** [reply]
You are right, this may be missing feature.
The command to update SDKs is the same as generating them:

```
yarn dlx @yarnpkg/sdks vscode
```

**Reply by rarkins** [reply]
We would still need to flesh out exactly how this should work. For example, should Renovate *ignore* the generated `package.json` files and instead rely on `yarn dlx` to manage them?
